lightningd/peer_htlcs.c

```diff
@@ -852,6 +852,7 @@ htlc_accepted_hook_deserialize(const tal_t *ctx,
 		fatal("Plugin responded with an unknown result to the "
 		      "htlc_accepted hook: %s",
 		      json_strdup(tmpctx, buffer, resulttok));
+		abort();
```

Comment:
This makes the code much uglier, and I can't tell *why* you need to do this since you didn't quote the error in the commit message (and, obviously, it doesn't happen here!).

fatal() is correctly labeled NORETURN; what is failing?

Reply:
Hi @rustyrussell , the reason I put in these `abort()`'s was that when I currently do `make check-source` on master d9b248241 it fails with:

```C
...
lightningd/peer_htlcs.c:900:9: error: Uninitialized variable: result [uninitvar]
 return result;
        ^
wallet/wallet.c:3744:4: error: Uninitialized variable: ann [uninitvar]
   ann->type = db_column_int(stmt, 8);
   ^
make: *** [Makefile:339: check-cppcheck] Error 123
```

When we look at the source `lightningd/peer_htlcs.c:900` or `wallet/wallet.c:3744`, we see that the code checker fails recognize that the previous command `fatal("Plugin responded ...")` already aborts or terminates in any circumstance, so that the uninitialized variable access to `result` or `ann` (wallet.c) are actual false positives. I decided to put in `abort()`'s , in order to not change the mechanics in how the variables are initialized or accessed so that the code-checks remain inplace. Maybe you know a nicer way for this.

Reply:
Maybe this is useful to you as well:
```
$> cppcheck --version 
Cppcheck 1.90
```

Reply:
I will change it to a cppcheck ignore comment like this: 
`// cppcheck-suppress autoVariables`

Reply:
Yes, please do.